flycheck-ledger.el

```diff
@@ -64,5 +65,72 @@ otherwise don't be pedantic."
 
 (add-to-list 'flycheck-checkers 'ledger)
 
+(defun flycheck-ledger-zero-last-position-of-account (account buffer)
+  "Return (LINE . COL) of last occurence of ACCOUNT in BUFFER.
+
+Return nil if ACCOUNT can't be found in BUFFER."
+  (with-current-buffer buffer
+    (save-restriction
+      (save-excursion
+        (goto-char (point-max))
+        (when (search-backward account nil t)
+          (cons (line-number-at-pos (point))
+                (1+ (- (point) (line-beginning-position)))))))))
+
+(defun flycheck-ledger-zero-error-parser (output checker buffer)
+  "Return errors found in OUTPUT.
+
+CHECKER is a `flycheck-ledger-zero' checker.
+
+_BUFFER is ignored.
+
+Return a list of parsed errors and warnings (as `flycheck-error'
+objects)."
+  (let ((errors (list))
+        (buffer (current-buffer)))
+    (save-match-data
+      (with-temp-buffer
+        (insert output)
+        (goto-char (point-min))
+        (while (re-search-forward "^ *\\(.*\\)  \\(.*\\)$" nil t)
+          (let* ((amount (match-string-no-properties 1))
+                 (account (match-string-no-properties 2))
+                 (message (format "Account %s should have zero value but has %s"
+                                  account amount))
+                 (position (flycheck-ledger-zero-last-position-of-account account buffer))
+                 (line (or (car position) 1))
+                 (column (or (cdr position) 0)))
+            (push
+             (flycheck-error-new
```

Comment:
Can be more succinct with `flycheck-error-new-at`.

Reply:
done